### DISCUSSION TITLE
Device not in `nanoff --listdevices` result

### DISCUSSION THREAD

**Discussion by regis1512** [nanoframework/Home#1454] [General]
### Tool

nanoff

### Description

My device from "Freenove Super Starter Kit for ESP32-S3" is not found in the list of devices

### How to reproduce

1. I execute 
`nanoff --target ESP32_S3_BLE --serialport COM5 --update --reset --masserase`
2. I get this result : 

> .NET nanoFramework Firmware Flasher v2.5.59+e3ff02e768
> Copyright (C) 2019 .NET Foundation and nanoFramework project contributors
> 
> 
> Reading details from chip...OK
> 
> Connected to:
> ESP32-S3 (ESP32-S3 (revision v0.1))
> Features WiFi, BLE
> Flash size 8MB GD25Q64 from GIGADEVICE (manufacturer 0x200 device 0x16407)
> PSRAM: undetermined
> Crystal 40MHz
> MAC 34:85:18:6D:35:40
> 
> Extracting ESP32_S3_BLE-1.9.1.21.zip...OK
> 
> Updating to 1.9.1.21
> 
> Erasing flash...OK
> Flashing firmware...OK

3. I execute 
`nanoff --listdevices`
4. I get this result

> .NET nanoFramework Firmware Flasher v2.5.59+e3ff02e768
> Copyright (C) 2019 .NET Foundation and nanoFramework project contributors
> 
> 
> No devices found

### Expected behaviour

I expect that the device is in the list
(I suppose it is why the device do no appear in the devices list in Visual Studio)

### Aditional context

I use : 
- Visual Studio : 2022
- .NET nanoFramework Extension : 2022.3.0.73
- target name : ESP32_S3_BLE
- firmware version : 1.9.1.21
- nanooff : 2.5.59

**Comment by regis1512** [comment]
I have follow the instructions in https://docs.nanoframework.net/content/getting-started-guides/troubleshooting-device-connection.html#finding-ports-on-windows

I have got ; 

![image](https://github.com/nanoframework/Home/assets/6832307/a7038b0e-a99a-40dc-9aed-c59921986864)

and : 

![image](https://github.com/nanoframework/Home/assets/6832307/ea87657c-86dd-468d-8f7f-907b051ab34f)

So now, i do not know if the problem come from nanoff or from the nano framework image which is not compatible with the board or something else.

**Comment by alberk8** [comment]
@regis1512 in the first screen capture, does the message run till the end.  On the same image is this the with a different baud rate?

**Comment by regis1512** [comment]
The first screen capture is Putty with a baud rate of 115200 (like indicated at https://docs.nanoframework.net/content/getting-started-guides/troubleshooting-device-connection.html#special-esp32). The message run till the end : when i do the screen capture there is no more information which appear in Putty since many seconds.

The second screen capture is Putty with the baud rate of 921600 (like indicated at https://docs.nanoframework.net/content/getting-started-guides/troubleshooting-device-connection.html#finding-if-nanoframework-booted). The message run till the end too.

**Comment by regis1512** [comment]
I have unplugged and plugged the cable many times.
I have tried other cables too.
Thanks.

**Comment by alberk8** [comment]
Does it work when you flash the non BLE firmware?  One more question, If you flash the using the USB on the right.  Unplug and plug in to the USB on the left and then VS should be able to view the device.

**Comment by josesimoes** [comment]
This is most likely the common situation of having it connected on the "wrong" USB connector.
I suppose that board of yours has a second USB connector? Please try connecting it there.
Also it would be helpful to include in the issue a link to the product page and/or schematic.

**Comment by josesimoes** [comment]
Moving this to a discussion as we don't use issues to answer questions and this is not a bug report, as noted in the issue notes. 😉

**Comment by MartinDavidWaller** [comment]
Hi, 

I was wondering if this problem was ever sorted. I have the same issue. I managed to download code a couple of times and then it went into a state where I can no longer see the device. I've tries --masserase, I've also reload Arduino scetches to it and then tried nanoff again and it still does not appear. Any advice much appreciated.

Martin

**Comment by alberk8** [comment]
Try this

I think it could be issue of partition size change that is causing the problem.
Try this.

go to this directory `C:\Users\username\.dotnet\tools\.store\nanoff\2.5.61\nanoff\2.5.61\tools\net6.0\any\esptool\esptoolWin
`
`esptool --chip esp32 erase_flash`

then
`nanoff --clearcache`

`nanoff --platform esp32 --serialport COM5 --update --masserase`

**Reply by WitchNectar** [reply]
This appears to have solved the problem for at least one of my ESP32-DevKitC's (ESP32-WROOM-32U; see my story below). It now appears when I nanoff --listdevices and in VS Device Explorer, and I was able to deploy code changes to the chip and test. Surprisingly, icons also reappeared in Device Explorer. I'm going to try my other boards next. Any idea why Device Explorer would just spontaneously quit working? What is being cached that would cause a failure without me doing anything? It sure seems like Device Explorer could be made smarter to detect connection and caching issues and give the option to clear the cache and reflash, or just cleanup automatically to avoid bricking everything it touches. Thank you, much appreciated. 🫡 I hope it keeps working. 🤞🏻

**Reply by alberk8** [reply]
One thing I know is that the Device Explorer will not brick your ESP32.  One of the reason for VS to not show up could be the USB cable, it happened to a few users as reported in nF Discord Server.   The other is if you have older firmware before the IDF 5.1.x  version.  The cause was the partition size was changed in recently to accommodate Matter Thread.   The effective way to clear the old partition is running the command below.  The nanoff or arduino does not reformat ESP32 partitions if one already exists.  Hope this helps to explain the issue.

`C:\Users\username\.dotnet\tools\.store\nanoff\2.5.90\nanoff\2.5.90\tools\net8.0\any\esptool\esptoolWin>esptool --chip esp32 erase_flash
`

**Comment by WitchNectar** [comment]
I have the same exact problem but with ESP32-DevKitC (ESP32-WROOM-32U). In fact, everything was working fine, until leaving the board running and plugged into USB overnight, then it disappeared from VS Device Explorer and would no longer appear with nanoff --listdevices. So I tried reflashing the board with nanoff as I did originally. It flashes successfully, proving that the physical USB port, COM port, and Windows driver are working, but then neither nanoff nor VS Device Explorer will recognize it. The board still works, because it's still running my code and performing as expected. I tried reinstalling the Silicon Labs USB to UART driver, reinstalling Visual Studio 2022 Community Edition, reinstalling nanoFramework, updating all the nanoFramework libraries with NuGet, creating a new nanoFramework project from scratch, switching to a virgin Windows 11 PC and installing everything from scratch, using various different USB cables and physical USB ports, and switching out the ESP32 board for an identical one. The only thing that worked was switching boards... for about a day. Then the new board was no longer recognized the following day after leaving it running overnight. I didn't change anything. Didn't unplug or reset, just wanted to change code and deploy it, but it disappeared from Device Explorer. So I tried nanoff --listdevices from a command prompt. Nope. Tried reflashing the firmware again (after updating nanoff again), which succeeds, but nanoff --listdevices and VS Device Explorer both still fail. I tried flashing 2 other virgin boards, different models but both based on ESP32-C3, which succeeded in flashing but aren't recognized thereafter by nanoff or Device Explorer. I know how to enter boot mode to flash and reset afterward to enter run mode. I know how to unplug and replug for hard reset. I read all the tutorials and troubleshooting guides. I used PuTTY to check the serial ports. I can read device details from the chips. There is definitely some type of bug or design flaw in both nanoff and nanoFramework Device Explorer. No user should have to suffer through this nonsense. I've been a loyal .NET C# developer professionally for 15 years, and I *really* want to make nanoFramework work, so I can do all kinds of cool stuff using .NET on various MCUs (lots of projects in mind like aircraft instruments, avionics, Bitcoin Lightning nodes, decentralized social media mesh networks, etc), but the way things are going, I'm ready to throw it all in the trash, walk away, and never come back. If I don't get a satisfactory answer or fix to this problem soon, I'm out.

**Reply by alberk8** [reply]
I am not sure why you are getting the errors.  Can you change USB cable.  I have no problem with VS not recognizing ESP32.

**Comment by MartinDavidWaller** [comment]
Hi Folks, 

The command:

C:\Users\marti\.dotnet\tools\.store\nanoff\2.5.90\nanoff\2.5.90\tools\net8.0\any\esptool\esptoolWin>esptool --chip esp32 erase_flash

did it for me too. It's now working as expected.

Many thanks

Martin
